File: oofem-csharp/oofem-02/Element.cs
public class Element
{
    public double E;
    public double A;
    public Node Node1;
    public Node Node2;

    public Element(double e, double a, Node n1, Node n2)
    {
        this.E = e;
        this.A = a;
        this.Node1 = n1;
        this.Node2 = n2;
    }

    public double Length()
    {
        return (this.Node2.Position - this.Node1.Position).Norm(2);
    }

    public Vector D()
    {
        return 1 / this.Length() * (this.Node2.Position - this.Node1.Position);
    }

    public Vector T()
    {
        Vector e1 = this.D();
        return Vector.Build.DenseOfEnumerable((-e1).Concat(e1));
    }

    public Matrix K()
    {
        return this.E * this.A / this.Length() * Vector.OuterProduct(this.T(), this.T());
    }

    public override string ToString()
    {
        return $"{this.E,15:0}{this.A,15:0.0000##}{this.Length(),15:0.0000}";
    }

}

File: oofem-python/oofem/element.py
import numpy as np

from node import Node


class Element:

    def __init__(
        self,
        e: float,
        a: float,
        node1: Node,
        node2: Node,
    ):

        self.e = e
        self.a = a
        self.node1 = node1
        self.node2 = node2

    def length(self) -> float:

        difference = self.node2.position - self.node1.position

        return np.linalg.norm(difference)

    def d(self) -> np.ndarray:

        return (
            self.node2.position - self.node1.position
        ) / self.length()

    def t(self) -> np.ndarray:

        e1 = self.d()

        return np.concatenate((-e1, e1))

    def stiffness_matrix(self) -> np.ndarray:

        return (
            self.e
            * self.a
            / self.length()
            * np.outer(self.t(), self.t())
        )

    def dofs(self) -> np.ndarray:

        return np.concatenate(
            (self.node1.dofs, self.node2.dofs)
        )

    def displacement(self) -> np.ndarray:

        return np.concatenate(
            (
                self.node1.displacement,
                self.node2.displacement,
            )
        )

    def normal_force(self) -> float:

        return (
            self.e
            * self.a
            / self.length()
            * np.dot(self.t(), self.displacement())
        )

    def __str__(self) -> str:

        return (
            f"{self.e:15.0f}"
            f"{self.a:15.6f}"
            f"{self.length():15.4f}"
        )